### PR TITLE
[wgsl-in] Don't allow keywords to be used as identifiers

### LINKED ISSUE #1546: [wgsl-in] naga should not accept keywords as identifiers
The code `let sampler = 0;` is accepted by naga, but according to the spec, an identifier [must not have the same spelling as a keyword or as a reserved word](https://gpuweb.github.io/gpuweb/wgsl/#identifiers), and `sampler` [is a keyword](https://gpuweb.github.io/gpuweb/wgsl/#keyword-summary).

### PR DESCRIPTION
**Pull request by Gordon-F**
Fix #1546, #1481.

### CODE REVIEW

File: src/front/wgsl/keywords.rs [hunk 1]
```diff
@@ -0,0 +1,133 @@
+pub const RESERVED: &[&str] = &[
```
Comment (Gordon-F) [inline src/front/wgsl/keywords.rs:1]: I think back and front should use the same reserved list. Currently just copy-paste.

Comment (kvark) [reply src/front/wgsl/keywords.rs:1]: good point! We need a place to share them

File: tests/wgsl-errors.rs [hunk 2]
```diff
@@ -13,6 +13,7 @@ fn check(input: &str, snapshot: &str) {
                 diff::Result::Right(r) => println!("+{}", r),
             }
         }
+        println!("Raw output:\n{}", output);
```
Comment (kvark) [inline tests/wgsl-errors.rs:16]: debug stuff?

Comment (Gordon-F) [reply tests/wgsl-errors.rs:16]: Yep, easier copy-paste for tests actual\expected stuff.

File: tests/wgsl-errors.rs [hunk 3]
```diff
@@ -531,6 +531,114 @@ fn postfix_pointers() {
...
+
+"###,
+    );
+
+    // struct
+    check(
+        r#"
+            struct array {};
+        "#,
+        r###"error: Name ` array` is a reserved keyword
```
Comment (Gordon-F) [inline tests/wgsl-errors.rs:618]: Avoiding extra spaces would be nice

File: src/lib.rs [hunk 4]
```diff
@@ -209,6 +209,7 @@ mod arena;
 pub mod back;
 mod block;
 pub mod front;
+pub(crate) mod keywords;
```
Comment (kvark) [inline src/lib.rs:212]: `pub(crate)` doesn't make sense in this place. Just remove it, leave only `mod keywords`.

**Review by kvark — APPROVED** [review]

**Pull request merged**